CRM/Price/BAO/LineItem.php

```diff
@@ -454,9 +454,14 @@ public static function processPriceSet($entityId, $lineItem, $contributionDetail
           }
           // CRM-19094: entity_table is set to civicrm_membership then ensure
           // the entityId is set to membership ID not contribution by default
-          elseif ($line['entity_table'] == 'civicrm_membership' && !empty($line['entity_id']) && $line['entity_id'] == $contributionDetails->id) {
-            $membershipId = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipPayment', 'contribution_id', $line['entity_id'], 'membership_id');
-            $line['entity_id'] = $membershipId ? $membershipId : $line['entity_id'];
+          // we don't update entity table and entity id during update mode, so
+          // no need to worry about the value of $line['entity_id'] while updating.
+          elseif (!$update && $line['entity_table'] == 'civicrm_membership' && !empty($line['entity_id']) && $line['entity_id'] == $contributionDetails->id) {
```

Comment:
As per https://github.com/civicrm/civicrm-core/blob/master/CRM/Price/BAO/LineItem.php#L62, we unset the content of entity_id and entity_table if we are editing the line item. Hence avoiding the statement if we are in update mode.

Reply:
Just looked at another linked PR - https://github.com/civicrm/civicrm-core/pull/11816 and the only difference between this and the other is the `$update` addition in the if condition here which seems to be passing all the unit test failures.